extensions/Worker.Extensions.Http.AspNetCore/src/HttpDataModel/AspNetCoreHttpHeadersCollection.cs

```diff
@@ -38,7 +38,10 @@ public AspNetCoreHttpResponseHeadersCollection(HttpResponse request)
             // Even though we're replacing the response feature headers, we need to make sure
             // headers are written back to the original reference before sent to the client as
             // that reference is still used.
-            request.OnStarting(ProcessResponseStarting);
+            if (!request.HasStarted)
+            {
+                request.OnStarting(ProcessResponseStarting);
```

Comment:
If this has already started, should we manually call `ProcessResponseStarting`? Or what is the impact of `ProcessResponseStarting` never being called?